### PR TITLE
tests: expand hpx::search_n unit tests to match HPX algorithm test standards

### PR DESCRIPTION
**Pull request by aneek22112007-tech**
This PR expands the `search_n` tests to bring them in line with the rest of the algorithms test suite (like `search.cpp`).

The current `search_n.cpp` was pretty barebones. It didn't use `test_utils.hpp` and only tested a very small 6-element vector with `par`, which means the actual parallel partitioning code (`detail::search_n::parallel` and `cancellation_token` stuff) wasn't really being exercised.

I rewrote the test file to mirror `search.cpp`'s structure:
- Tests all policies (`seq`, `par`, `par_unseq`, `seq(task)`, `par(task)`) and iterator tags (random access and forward).
- Uses a 10,007-element vector to ensure the parallel partitioner actually splits work across threads.
- Covers C++20 edge cases (`count == 0`, `count > dist`, etc).
- Tests exception and `bad_alloc` propagation using `decorated_iterator`.
- Adds a test with a non-commutative predicate to catch any future argument-order bugs.
- Adds the standard `--seed` flag.

Just a straightforward test expansion, no changes to the algorithm implementation itself.

### CODE REVIEW

**Review by hkaiser — COMMENTED** [review]
Thanks for looking into increasing test coverage for this algorithm. Could you please have a look at the bot's comments? Either mark those as resolved with a comment of your own or address them.

File: libs/core/algorithms/tests/unit/algorithms/search_n.cpp [hunk 1]
```diff
@@ -9,87 +9,878 @@
...
+void test_search_n_bad_alloc(ExPolicy policy, IteratorTag)
+{
+    static_assert(hpx::is_execution_policy<ExPolicy>::value,
+        "hpx::is_execution_policy<ExPolicy>::value");
+
+    typedef std::vector<std::size_t>::iterator base_iterator;
+    typedef test::decorated_iterator<base_iterator, IteratorTag>
+        decorated_iterator;
+
+    std::vector<std::size_t> c(100007);
```
Comment (hkaiser) [inline libs/core/algorithms/tests/unit/algorithms/search_n.cpp]: Is this a typo (or a deliberate change)? Other tests use `10007` elements. Not a biggy, just asking.

Comment (aneek22112007-tech) [reply libs/core/algorithms/tests/unit/algorithms/search_n.cpp]: Yes, that was indeed a typo! Good catch. I've updated it to 10007 to be consistent with the rest of the tests, and also addressed the other comments regarding the dead fixture data and redundant 4b test case in the latest commit

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!
